services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImpl.java

```diff
@@ -1096,6 +1097,63 @@ public S3FileHandle createExternalS3FileHandle(UserInfo userInfo,
 		return (S3FileHandle) fileHandleDao.createFile(fileHandle);
 	}
 
+	@Override
+	public GoogleCloudFileHandle createExternalGoogleCloudFileHandle(UserInfo userInfo,
+												   GoogleCloudFileHandle fileHandle) {
+		ValidateArgument.required(userInfo, "userInfo");
+		ValidateArgument.required(fileHandle, "fileHandle");
+		ValidateArgument.required(fileHandle.getStorageLocationId(), "FileHandle.storageLocationId");
+		ValidateArgument.requiredNotEmpty(fileHandle.getBucketName(), "FileHandle.bucket");
+		ValidateArgument.requiredNotEmpty(fileHandle.getKey(), "FileHandle.key");
+		ValidateArgument.requiredNotEmpty(fileHandle.getContentMd5(),"FileHandle.contentMd5");
+		if (fileHandle.getFileName() == null) {
+			fileHandle.setFileName(NOT_SET);
+		}
+		if (fileHandle.getContentType() == null) {
+			fileHandle.setContentType(NOT_SET);
+		}
+
+		if (!MD5ChecksumHelper.isValidMd5Digest(fileHandle.getContentMd5())) {
+			throw new IllegalArgumentException("The content MD5 digest must be a valid hexadecimal string of length 32.");
+		}
+
+		// Lookup the storage location
+		StorageLocationSetting sls = storageLocationDAO.get(fileHandle.getStorageLocationId());
+		ExternalGoogleCloudStorageLocationSetting esls = null;
+		if(!(sls instanceof ExternalGoogleCloudStorageLocationSetting)){
+			throw new IllegalArgumentException("StorageLocationSetting.id="+fileHandle.getStorageLocationId()+" was not of the expected type: "+ExternalGoogleCloudStorageLocationSetting.class.getName());
+		}
+		esls = (ExternalGoogleCloudStorageLocationSetting) sls;
+		if(!fileHandle.getBucketName().equals(esls.getBucket())){
+			throw new IllegalArgumentException("The bucket for ExternalGoogleCloudStorageLocationSetting.id="+fileHandle.getStorageLocationId()+" does not match the provided bucket: "+fileHandle.getBucketName());
+		}
+
+		/*
+		 *  The creation of the ExternalGoogleCloudStorageLocationSetting already validates that the user has
+		 *  permission to update the bucket. So the creator of the storage location is
+		 *  the only one that can create an GoogleCloudFileHandle using that storage location Id.
+		 */
+		if(!esls.getCreatedBy().equals(userInfo.getId())){
+			throw new UnauthorizedException("Only the creator of ExternalGoogleCloudStorageLocationSetting.id="+fileHandle.getStorageLocationId()+" can create an external GoogleCloudFileHandle with storageLocationId = "+fileHandle.getStorageLocationId());
+		}
+		try {
+			Blob summary = googleCloudStorageClient.getObject(fileHandle.getBucketName(), fileHandle.getKey());
+			if (fileHandle.getContentSize() == null) {
+				fileHandle.setContentSize(summary.getSize());
+			}
+		} catch (Exception e) {
```

Comment:
either (1) replace 'Exception' with the particular exception type thrown by 'getObject' or (2) ensure that only the getObject() method invocation is within the try/catch block.